File: server/src/main/scala/kpn/api/custom/RouteMemberInfo.scala
package kpn.api.custom

import kpn.api.common.route.RouteNetworkNodeInfo
import kpn.api.common.route.WayDirection

case class RouteMemberInfo(
  id: Long,
  memberType: String,
  isWay: Boolean,
  nodes: Seq[RouteNetworkNodeInfo],
  linkName: String,
  from: String,
  fromNodeId: Long,
  to: String,
  toNodeId: Long,
  role: String,
  timestamp: Timestamp,
  isAccessible: Boolean,
  length: String,
  nodeCount: String,
  description: String,
  oneWay: WayDirection,
  oneWayTags: Tags
)

File: client/src/app/kpn/api/custom/route-member-info.ts
import { RouteNetworkNodeInfo } from '../common/route';
import { Tags } from './tags';
import { Timestamp } from './timestamp';

export interface RouteMemberInfo {
  readonly id: number;
  readonly memberType: string;
  readonly isWay: boolean;
  readonly nodes: RouteNetworkNodeInfo[];
  readonly linkName: string;
  readonly from: string;
  readonly fromNodeId: number;
  readonly to: string;
  readonly toNodeId: number;
  readonly role: string;
  readonly timestamp: Timestamp;
  readonly accessible: boolean;
  readonly length: string;
  readonly nodeCount: string;
  readonly description: string;
  readonly oneWay: string;
  readonly oneWayTags: Tags;
}
